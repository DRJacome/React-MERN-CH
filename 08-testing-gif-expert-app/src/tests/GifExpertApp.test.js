import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp'

describe('Pruebas sobre el componente <GifExpertApp/>', () => {

  test('Debe mostrar el componente correctamente', () => {
    const wrapper = shallow(<GifExpertApp />)

    expect(wrapper).toMatchSnapshot();
  })

  test('Debe mostrar una lista de categorÃ­as', () => {
    const categorias = ['One Punch Man', 'Kimetsu no Yaiba'];
    const wrapper = shallow(<GifExpertApp categoriaPorDefecto={categorias} />)

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categorias.length);
  })
})