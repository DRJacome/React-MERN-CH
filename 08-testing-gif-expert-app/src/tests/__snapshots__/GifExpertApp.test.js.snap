// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifExpertApp/> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    Gif Finder App
  </h2>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre el componente <GifExpertApp/> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    Gif Finder App
  </h2>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="One Punch Man"
      key="One Punch Man"
    />
    <GifGrid
      categoria="Kimetsu no Yaiba"
      key="Kimetsu no Yaiba"
    />
  </ol>
</Fragment>
`;
